tests/test_fetch.py

```
@@ -0,0 +1,23 @@
+# tests/test_fetch.py
+
+import pytest
+from unittest.mock import patch
+import pandas as pd
+
+
+def test_get_historical_weather_returns_dataframe():
+    """Verify that get_historical_weather returns a non-empty DataFrame with temperature data."""
+    lat = 55.9533  # Edinburgh
+    lon = -3.1883
+    start = datetime(2023, 1, 1)
```

Ruff (0.11.9)

Undefined name `datetime`

(F821)